backend/app/src/main/java/com/moh/phlat/backend/repository/ProcessDataRepository.java

```diff
@@ -33,4 +35,413 @@ public interface ProcessDataRepository extends JpaRepository<ProcessData, Long>
 	long countAllByControlTableIdAndRowstatusCode(Long controlTableId, String string);
 
 	long countByControlTableId(Long controlTableId);   
+	
+public List<ProcessData> findAll(Specification<ProcessData> spec);
+	
+	@Query(value="Select distinct id FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
```

Comment:
There should be some way like using specification to generalize instead of using so many methods in repo. 
Also in all the methods, any reason why flat nativeQuery is set to true?

Reply:
I was trying different ways to use a single method, but they kept failing.  The nativeQuery came from these attempts and I didn't test to see if it still worked after the current iteration.  I will look into whether specification can be used.